Add support for LocalConnectTimeoutMs and LocalRequestTimeoutMs on service-defaults CRD

erdanzhang opened the pull request:
Note original PR https://github.com/hashicorp/consul-k8s/pull/1642, but I couldn't get CLA works since I signed it with a different email than the author's, and seems cannot be changed.

Changes proposed in this PR:

Add maxInboundConnections to service-defaults CRD
fix https://github.com/hashicorp/consul-k8s/issues/1641
Checklist:

Tests added
Reference of similar PR before adding entries to serviceDefault:
https://github.com/hashicorp/consul-k8s/pull/1437

hashicorp-cla commented:
[![CLA assistant check](https://cla.hashicorp.com/pull/badge/not_signed)](https://cla.hashicorp.com/hashicorp/consul-k8s?pullRequest=1647)


Thank you for your submission! We require that all contributors sign our Contributor License Agreement ("CLA") before we can accept the contribution. [Read and sign the agreement](https://cla.hashicorp.com/hashicorp/consul-k8s?pullRequest=1647)

[Learn more about why HashiCorp requires a CLA and what the CLA includes](https://www.hashicorp.com/cla)

<sub>Have you signed the CLA already but the status is still pending? [Recheck it](https://cla.hashicorp.com/check/hashicorp/consul-k8s?pullRequest=1647).</sub>

erdanzhang commented:
@thisisnotashwin Per your comments in the original PR:
```
The changes to this file should be autogenerated by running make ctrl-generate ctrl-manifests
```

Auto-gen code is cool.
After I ran make ctrl-generate ctrl-manifests, there were few other files that got changed as well.
```
	modified:   charts/consul/templates/crd-peeringacceptors.yaml
	modified:   charts/consul/templates/crd-peeringdialers.yaml
	modified:   charts/consul/templates/crd-servicedefaults.yaml
	modified:   control-plane/api/v1alpha1/zz_generated.deepcopy.go
	modified:   control-plane/config/crd/bases/consul.hashicorp.com_servicedefaults.yaml
```
Could you help double-check if the changes are intended?

thisisnotashwin commented on charts/consul/templates/crd-peeringacceptors.yaml:
```diff
@@ -1,4 +1,4 @@
-{{- if and .Values.connectInject.enabled .Values.global.peering.enabled }}
+{{- if .Values.controller.enabled }}
```
Can this be reverted from this file and the one below it?

t-eckert replied:
+1

erdanzhang replied:
@thisisnotashwin Thanks for taking the time to review.
I reverted the change to 
```
charts/consul/templates/crd-peeringacceptors.yaml
charts/consul/templates/crd-peeringdialers.yaml
```

And kept the change to 

```
	modified:   charts/consul/templates/crd-servicedefaults.yaml
	modified:   control-plane/api/v1alpha1/zz_generated.deepcopy.go
	modified:   control-plane/config/crd/bases/consul.hashicorp.com_servicedefaults.yaml
```

For file, control-plane/api/v1alpha1/zz_generated.deepcopy.go, not sure if the change looks good
<img width="1053" alt="image" src="https://user-images.githubusercontent.com/84437068/200218950-4da166a6-6ce5-4870-9066-75335201cea0.png">

Could you help confirm if the changes are correct?

thisisnotashwin replied:
Hey @erdanzhang !! Those changes look right!! I really appreciate you reaching out to confirm it!! Also, thank you so much, again, for your contribution!!

thisisnotashwin reviewed (commented):
This looks great! Just holding off of the approval as we need to revert the `if` conditions on the acceptors and dialers yaml files.

t-eckert reviewed (approved):
Excellent work! Thank you for doing this!
